File: app/src/main/java/pt/ipbeja/pi/piproject/persistence/IdentificationDao.java
package pt.ipbeja.pi.piproject.persistence;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

/**
 * Data access object for identifications
 */
@Dao
public interface IdentificationDao {
    /**
     * Inserts a record of an identification in the database
     *
     * @param identification the identification to insert
     */
    @Insert
    public void insertIdentification(Identification identification);

    /**
     * Removes an identification from the the database
     *
     * @param identification the identification to remove
     */
    @Delete
    public void deleteIdentification(Identification identification);

    @Update
    public void updateOrderInIdentification(Identification identification);

    /**
     * Retrieves all the identifications in the database
     *
     * @return all the identifications in the database
     */
    @Query("SELECT * FROM identification")
    public Identification[] getAllIdentifications();

    @Query("SELECT COUNT(*) FROM identification")
    public int getIdentificationCount();
}

File: app/src/main/java/pt/ipbeja/pi/piproject/persistence/IdentificationDao.kt
package pt.ipbeja.pi.piproject.persistence

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

/**
 * Data access object for identifications
 */
@Dao
interface IdentificationDao {
    /**
     * Inserts a record of an identification in the database
     *
     * @param identification the identification to insert
     */
    @Insert
    fun insertIdentification(identification: Identification?)

    /**
     * Removes an identification from the the database
     *
     * @param identification the identification to remove
     */
    @Delete
    fun deleteIdentification(identification: Identification?)

    @Update
    fun updateOrderInIdentification(identification: Identification?)

    @get:Query("SELECT * FROM identification")
    val allIdentifications: Array<Identification?>?

    @get:Query("SELECT COUNT(*) FROM identification")
    val identificationCount: Int
}